sentry_sdk/utils.py

```diff
@@ -891,21 +891,31 @@ def __init__(self, waiting_time, configured_timeout):
         threading.Thread.__init__(self)
         self.waiting_time = waiting_time
         self.configured_timeout = configured_timeout
+        self._stop_event = threading.Event()
+
+    def stop(self):
+        # type: () -> None
+        self._stop_event.set()
+
+    def stopped(self):
+        # type: () -> Any
+        return self._stop_event.is_set()
 
     def run(self):
         # type: () -> None
 
         time.sleep(self.waiting_time)
```

Comment:
The thread doesn't need to sleep for the full `waiting_time`. When the handler returns and we call `timeout_thread.stop()`, the thread should wake up and terminate.

That's why I suggested using `self._stop_event.wait(self.waiting_time)`.

https://docs.python.org/2.7/library/threading.html#threading.Event.wait

You can either use the return value or `wait` or `.is_set()` to decide whether to raise `ServerlessTimeoutWarning`.

Reply:
Yes I figured that while I did testing and found it was still waiting for complete waiting time. 
I've made final changes accordingly now.